### PR TITLE
PCCTC Project Template

### PR DESCRIPTION
**Pull request by ddsjoberg**
- [x] add `use_file_*()` functions
- [x] add unit tests
- [x] Should i add specific medidata setup file with a second example for analysis data sets
- [ ] need to think of a structure for saving analytic databases and results. probably not the same place as the raw data.
- [x] do i need to have a function like `here_data()`
- [x] added a function for labelling the variables...does our excel file need additional information?
- [ ] add code to label variables to the medidata setup template
- [ ] should raw variables all be lower case since CDISC has upper case derived vars?
- [x] update template files to attach the data science umbrella pkg
- [ ] Replace calls to `mskRvis::theme_biostat()` with something else
- [ ] add a template file for labelling the cleaned medidata data

### CODE REVIEW

**Comment by ddsjoberg** [comment]
@shannonpileggi here's the first draft of the template for projects. What do you think ?

this may also be somewhat helpful
https://github.mskcc.org/pages/pcctc/data-science-guide/102-project_setup.html

File: inst/project-templates/setup.qmd [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+```{r checking}
+
+```
+
+# Save Analytic Data
+
+```{r save}
+# applying variable labels, and dropping raw variables
+df_main <- df_main %>%
+  set_variable_labels(path = here::here("scripts/derived_variables_{{stringr::str_split(folder_name, pattern = ' |-', simplify = T)[, 1] %>% tolower()}}.xlsx"))
```
Comment (shannonpileggi) [inline inst/project-templates/setup.qmd]: update with new function name and location

File: inst/project-templates/setup.qmd [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# setting MSK ggplot theme, gtsummary theme, and color palette
+theme_set(mskRvis::theme_biostat()); theme_gtsummary_pcctc(); mskRvis::set_msk_palette()
+
+# example code for importing excel file
+df_raw_data <-
+  readxl::read_excel(
+    path = here_data("Raw Data from PI.xlsx")
+  ) %>%
+  janitor::clean_names(case = "all_caps") %>% # uppercase col names
+  mutate(across(where(is.POSIXt), as_date)) # use lubridates
```
Comment (shannonpileggi) [inline inst/project-templates/setup.qmd:38]: we could add in read_sas_batch

Comment (shannonpileggi) [reply inst/project-templates/setup.qmd:38]: read_sas_batch could be enhanced with (1) data processing, and (2) DRS labels

Comment (shannonpileggi) [reply inst/project-templates/setup.qmd:38]: experiment w/ zipped files in temp directory

File: inst/project-templates/setup-medidata.qmd [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+
+```{r import}
+library(dsverse); library(tidyverse)
+# library(tidylog)
+
+# setting MSK ggplot theme, gtsummary theme, and color palette
+theme_set(mskRvis::theme_biostat()); theme_gtsummary_pcctc(); mskRvis::set_msk_palette()
+
+# import sas files
+lst_medidata_raw_data <-
```
Comment (shannonpileggi) [inline inst/project-templates/setup-medidata.qmd:39]: oooh i see you have a separate script for medidata set up. i think i started the project with medidata FALSE to see how it went.  often analyses need a bespoke analysis file & medidata - should this be one script, or kept separate? (and different deliverables will have different bespoke excel files)

File: inst/project-templates/setup-medidata.qmd [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+```
+
+# Save Analytic Data
+
+```{r save}
+#saving cleaned Medidata files
+save(
+  # save all cleaned medidata data frames
+  list =
```
Comment (shannonpileggi) [inline inst/project-templates/setup-medidata.qmd]: let's talk about what this does

**Review by shannonpileggi — COMMENTED** [review]
Do we also want to include parameterized quarto rendering for dated output files?

File: inst/project-templates/setup-medidata.qmd (hunk 4 above)
Comment (ddsjoberg) [reply inst/project-templates/setup-medidata.qmd]: update template to save out a list of checked medidata dfs

File: inst/project-templates/setup-medidata.qmd (hunk 3 above)
Comment (ddsjoberg) [reply inst/project-templates/setup-medidata.qmd:39]: as discussed, we will keep medidata separate from other setups

File: inst/project-templates/setup.qmd (hunk 1 above)
Comment (ddsjoberg) [reply inst/project-templates/setup.qmd]: done!

File: inst/project-templates/setup-medidata.qmd (hunk 4 above)
Comment (ddsjoberg) [reply inst/project-templates/setup-medidata.qmd]: updated!

**Pull request merged**